pkg/util/utils_linux.go

```diff
@@ -48,7 +48,16 @@ func FindDeviceNodes() (map[string]string, error) {
 
 		info, err := d.Info()
```

Comment:
A few lines up (line 45) there's `d.Type()`, doesn't that require a `stat()` also? Sorry for the drive-by comment, I've tried to read the `go doc` but it's not as helpful as I need.

Reply:
No that is one of the reasons go introduced `WalkDir()` vs the old `Walk()`. If you read the directory entries via `getdents()` system call the structure will include the file type so there is no need for a stat on each file.
Walking is just a simple open(), getdents(), close(). The fs.DirEntry interface exposes basically parts of the linux_dirent structure. Only if we ask for full info it needs to stat.